app/src/main/java/com/celzero/bravedns/util/OrbotHelper.kt

```diff
@@ -363,41 +367,38 @@ class OrbotHelper(private val context: Context, private val persistentState: Per
 
         socks5Port = socks5ProxyPort as Int?
         httpsPort = httpsProxyPort as Int?
-        socks5IP = socks5ProxyHost as String?
-        httpsIP = httpsProxyHost as String?
+        socks5Ip = socks5ProxyHost as String?
+        httpsIp = httpsProxyHost as String?
 
         if (DEBUG) Log.d(LOG_TAG_VPN,
-                         "OrbotHelper - Orbot - $socks5Port, $httpsPort, $socks5IP, $httpsIP")
+                         "OrbotHelper - Orbot - $socks5Port, $httpsPort, $socks5Ip, $httpsIp")
     }
 
     /**
      * Create a ScheduledExecutorService which will be executed after 30 sec of Orbot status
      * initiation.
      */
-    private fun timeOutForOrbot() {
-        // Create an executor that executes tasks in a background thread.
-        val backgroundExecutor: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
-
-        // Execute a task in the background thread after 30 sec.
-        // If there is no response for the broadcast from the Orbot,
-        // then the executor will execute and will disable the Orbot settings.
-        // Some cases where the Orbot won't be responding -eg., force stop of application,
-        // user disabled auto-start of the application.
-        backgroundExecutor.schedule({
-                                        Log.i(LOG_TAG_VPN,
-                                              "timeOutForOrbot executor triggered - $isResponseReceivedFromOrbot")
-                                        if (!isResponseReceivedFromOrbot) {
-                                            stopOrbot(isInteractive = false)
-                                        }
-                                        backgroundExecutor.shutdown()
-                                    }, 25, TimeUnit.SECONDS)
+    private suspend fun waitForOrbot() {
+        uiCtx {
+            delay(TimeUnit.SECONDS.toMillis(25L))
```

Comment:
the caller would wait for 25s, but that's undesirable since this method is called from `.startOrbot` which is inturn called from brave-vpn-service main-thread `onStartCommand`.

call this ui-coroutine in a separate launch

Reply:
calling in a separate launch